core/src/test/java/org/elasticsearch/common/xcontent/XContentParserUtilsTests.java

```diff
@@ -63,23 +64,13 @@ public void testParseTypedKeysObject() throws IOException {
         }));
         final NamedXContentRegistry namedXContentRegistry = new NamedXContentRegistry(namedXContents);
 
-        BytesReference bytes = toXContent((builder, params) -> builder.field("test", 0), xContentType, randomBoolean());
-        try (XContentParser parser = xContentType.xContent().createParser(namedXContentRegistry, bytes)) {
-            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
-            ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.nextToken(), parser::getTokenLocation);
-
-            ParsingException e = expectThrows(ParsingException.class, () -> parseTypedKeysObject(parser, delimiter, Boolean.class));
-            assertEquals("Cannot parse object of class [Boolean] without type information. Set [typed_keys] parameter " +
-                    "on the request to ensure the type information is added to the response output", e.getMessage());
-        }
```

Comment:
The test for the exception isn't valid any more. We could check that the consumer doesn't get called in this scenario. I can change that.